src/realm/db.cpp

```diff
@@ -2293,7 +2299,18 @@ VersionID Transaction::commit_and_continue_as_read(bool commit_to_disk)
     // Grabbing the new lock before releasing the old one prevents m_transaction_count
     // from going shortly to zero
     db->grab_read_lock(new_read_lock, version_id); // Throws
-    db->release_read_lock(m_read_lock);
+
```

Comment:
Could this logic be rewritten so it's easier to understand?

Reply:
I did try several other constructs, but they were all harder to understand or wrong. I will add some (more) comments